nion/swift/LinePlotCanvasItem.py

```diff
@@ -430,35 +432,66 @@ def __view_to_intervals(self, data_and_metadata: DataAndMetadata.DataAndMetadata
         else:
             left = right = 0.0
 
-        data_size = int(data_and_metadata.data_shape[-1])
-        left_channel = int(numpy.floor(left * data_size))
-        right_channel = int(numpy.ceil(right * data_size))
-        if left_channel > right_channel:
-            right_channel, left_channel = left_channel, right_channel
-        if left_channel not in range(data_size):
-            left_channel = 0
-        if right_channel not in range(data_size):
-            right_channel = data_size
-
-        channel_data = data_and_metadata.data
-        assert channel_data is not None
-        if channel_data[..., left_channel:right_channel].size > 0:
-            data_min = numpy.amin(channel_data[..., left_channel:right_channel])
-            data_max = numpy.amax(channel_data[..., left_channel:right_channel])
-        else:
-            data_min = numpy.amin(channel_data[..., :])
-            data_max = numpy.amax(channel_data[..., :])
-
-        y_min = 0.0 if data_min > 0 else data_min * 1.2
-        y_max = 0.0 if data_max < 0 else data_max * 1.2
+        # Isolate static properties (at draw-time) of top-level layer
+        top_layer_width = data_and_metadata.data_shape[-1]
+        top_layer_x_offset = data_and_metadata.dimensional_calibrations[0].offset
+        top_layer_x_scale = data_and_metadata.dimensional_calibrations[0].scale
+        top_layer_y_scale = data_and_metadata.intensity_calibration.scale
+        top_layer_y_offset = data_and_metadata.intensity_calibration.offset
+
+        selected_y_minimums: typing.List[float] = []
+        selected_y_maximums: typing.List[float] = []
+        global_y_minimums: typing.List[float] = []
+        global_y_maximums: typing.List[float] = []
+
+        for stacked_data_layer in self.__xdata_list:
```

Comment:
Overall, this algorithm is confusing. Can you write an explanation of how it is supposed to work? For instance, why do you have selected and global min/max arrays? What are the units of the min/max values stored in either array? What is the purpose of the `if layer_values.size <= 0:` line? etc.

Also, assigning a variable (y_min/maximums) to one of the the two arrays (selected_y_min/maximums vs global_y_min/maximums) and then appending values to it is what I would consider "tricky" code. Can you rewrite that to be more straightforward?